### PR TITLE
Adds PHP 8.3 as new supported version

### PR DESCRIPTION
**Pull request by fezfez**
replace https://github.com/laminas/laminas-filter/pull/109

### CODE REVIEW

**Comment by gsteel** [comment]
@fezfez - If you're on a Mac, try setting/updating the baseline using `--threads=1` - this _should_ make the baseline/issues even with the CI container

**Comment by fezfez** [comment]
> --threads=1

im on ubuntu, and i dont have the same error on ci...

File: test/Compress/TarTest.php [hunk 1]
```diff
@@ -185,6 +186,10 @@ public function testTarCompressDirectory(): void
 
     public function testSetModeShouldWorkWithCaseInsensitive(): void
     {
+        if (! extension_loaded('bz2')) {
+            self::markTestSkipped('This adapter needs the bz2 extension');
+        }
+
```
Comment (gsteel) [inline test/Compress/TarTest.php:192]: Why is this necessary? `Archive_Tar` does not _require_ Bz2 as far as I know, although it _can_ use it when available.

File: test/Word/CamelCaseToSeparatorNoPcreUnicodeTest.php [hunk 2]
```diff
@@ -5,28 +5,42 @@
...
     {
-        $this->reflection->setValue(true);
+        if (PHP_VERSION_ID >= 80300) {
+            $reflectionClass = new ReflectionClass(StringUtils::class);
+            $reflectionClass->setStaticPropertyValue('hasPcreUnicodeSupport', true);
+            return;
+        }
+
+        $reflection = new ReflectionProperty(StringUtils::class, 'hasPcreUnicodeSupport');
+        $reflection->setValue(true);
```
Comment (gsteel) [inline test/Word/CamelCaseToSeparatorNoPcreUnicodeTest.php]: I don't understand the version check here - `setStaticPropertyValue` has been available since version 5.1

**Review by gsteel — COMMENTED** [review]
Couple of questions @fezfez

File: test/Compress/TarTest.php (hunk 1 above)
Comment (fezfez) [reply test/Compress/TarTest.php:192]: because of `foreach (['GZ', 'Bz2'] as $mode) {` line 196

File: test/Word/CamelCaseToSeparatorNoPcreUnicodeTest.php (hunk 2 above)
Comment (fezfez) [reply test/Word/CamelCaseToSeparatorNoPcreUnicodeTest.php]: my bad :)

**Review by gsteel — APPROVED** [review]
Thanks @fezfez 👍

**Comment by froschdesign** [comment]
@fezfez @gsteel 
Thank you for your time, this contribution and your help! 👍
